eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ExternalDependencyExposedCheck.java

```diff
@@ -0,0 +1,236 @@
+        }
+    }
+
+    /**
+     * Get invalid return types from a given TYPE node.
+     *
+     * @param typeToken TYPE AST node
+     * @return a map that maps the invalid TYPE node and the type name.
+     */
+    private Map<DetailAST, String> getInvalidReturnTypes(DetailAST typeToken) {
+        Map<DetailAST, String> invalidReturnTypeMap = new HashMap<>();
+        final int typeCount = typeToken.getChildCount();
+        // The TYPE node could has more than one child, such as Map<T, T>
+        if (typeCount == 1) {
+            final String returnTypeName = typeToken.getFirstChild().getText();
+            // if not exist in the classPathMap, that implies the type is java default types, such as int.
+            if (classPathMap.containsKey(returnTypeName) && !isValidLibrary(classPathMap.get(returnTypeName))) {
+                invalidReturnTypeMap.put(typeToken, returnTypeName);
+            }
+        } else {
+            invalidReturnTypeMap = getInvalidTypeFromTypeArguments(typeToken);
+        }
+        return invalidReturnTypeMap;
+    }
+
+    /**
+     * Get invalid parameter types from a given PARAMETERS node.
+     *
+     * @param parametersTypeToken PARAMETERS AST node
+     * @return a map that maps all the invalid TYPE_ARGUMENT node and the type name
+     */
+    private Map<DetailAST, String> getInvalidParameterTypes(DetailAST parametersTypeToken) {
+        final Map<DetailAST, String> invalidParameterTypesMap = new HashMap<>();
+        for (DetailAST ast = parametersTypeToken.getFirstChild(); ast != null; ast = ast.getNextSibling()) {
+            if (ast.getType() == TokenTypes.PARAMETER_DEF) {
+                invalidParameterTypesMap.putAll(getInvalidTypeFromTypeArguments(ast.findFirstToken(TokenTypes.TYPE)));
+            }
+        }
+        return invalidParameterTypesMap;
+    }
+
+    /**
+     *  A helper function that checks TYPE AST node. Since both return type and input parameter argument type has
+     *  TYPE AST node under. This function applied to both.
+     *
+     * @param typeToken TYPE AST node
+     * @return a map that maps all the invalid TYPE_ARGUMENT node and the type name
+     */
+    private Map<DetailAST, String> getInvalidTypeFromTypeArguments(DetailAST typeToken) {
+        final Map<DetailAST, String> invalidTypesMap = new HashMap<>();
+
+        if (typeToken == null) {
+            return invalidTypesMap;
+        }
+
+        final DetailAST identToken = typeToken.findFirstToken(TokenTypes.IDENT);
+        // if there is no IDENT token, implies the token is a default java type.
+        if (identToken == null) {
+            return invalidTypesMap;
+        }
+
+        final String typeName = identToken.getText();
+        // if does not exist in the classPathMap, that implies the type is java default types, such as int.
+        if (classPathMap.containsKey(typeName)) {
```

Comment:
Can you reduce the number of nested ifs here? I can see two cases where you can fail out of this method early.